File: gateway/src/auth/dto/register.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEmail } from 'class-validator';

export class RegisterDTO {
  @ApiProperty({ required: true })
  @IsEmail()
  email: string;

  @ApiProperty({ required: true })
  password: string;

  @ApiProperty({ required: true })
  name: string;
}

File: auth-service/auth-service/DTO/RegisterDto.cs
using System.ComponentModel.DataAnnotations;

namespace auth_service.DTO
{
    public class RegisterDto
    {
        [EmailAddress]
        public string email {  get; set; }
        [Required]
        public string password { get; set; }
        public string name { get; set; }
    }
}
